docs/content/integrations/dbt/using-dbt-with-dagster/downstream-assets.mdx

```diff
@@ -13,119 +13,112 @@ description: Dagster can orchestrate dbt alongside other technologies.
+
+2. Before `defs = Definitions(`, add the definition for the `order_count_chart` asset:
 
-   ```python
-   @asset(ins={"customers": AssetIn(key_prefix=["jaffle_shop"])})
-   def order_count_chart(context, customers: pd.DataFrame) -> None:
+   ```python file=integrations/dbt/tutorial/downstream_assets/definitions.py startafter=start_downstream_asset endbefore=end_downstream_asset
+   @asset(deps=get_asset_key_for_model([jaffle_shop_dbt_assets], "customers"))
+   def order_count_chart(context):
+       # read the contents of the customers table into a Pandas DataFrame
+       connection = duckdb.connect(os.fspath(duckdb_database_path))
+       customers = connection.sql("select * from customers").df()
+
+       # create a plot of number of orders by customer and write it out to an HTML file
        fig = px.histogram(customers, x="number_of_orders")
        fig.update_layout(bargap=0.2)
-       save_chart_path = file_relative_path(__file__, "order_count_chart.html")
+       save_chart_path = duckdb_database_path.parent.joinpath("order_count_chart.html")
        fig.write_html(save_chart_path, auto_open=True)
 
-       context.add_output_metadata({"plot_url": MetadataValue.url("file://" + save_chart_path)})
+       # tell Dagster about the location of the HTML file, so it's easy to access from the Dagster UI
+       context.add_output_metadata(
+           {"plot_url": MetadataValue.url("file://" + os.fspath(save_chart_path))}
+       )
    ```
 
-Let's review what's happening here:
-
-- `customers` is supplied as an argument to `ins`, defining it as an upstream asset dependency of the `order_count_chart` asset
-
-- We've also used <PyObject object="AssetIn" /> to explicitly define the key prefix of the upstream asset we're referencing, because, earlier, we assigned this key prefix to all our dbt models.
-
-- Finally, the chart is saved as `order_count_chart.html` in `/tutorial_template/tutorial_dbt_dagster` and automatically opened in the browser upon successful materialization.
-
-At this point, `/tutorial_template/tutorial_dbt_dagster/assets/__init__.py` should look like this:
-
-```python
-import pandas as pd
-import plotly.express as px
-from dagster_dbt import load_assets_from_dbt_project
+   This asset definition looks similar the asset we defined in the previous section. In this case, instead of fetching data from an external source and writing it to DuckDB, it reads data from DuckDB, and then uses it to make a plot.
 
-from dagster import AssetIn, MetadataValue, asset, file_relative_path
+   The line `deps=get_asset_key_for_model([jaffle_shop_dbt_assets], "customers")` tells Dagster that this asset is downstream of the `customers` dbt model. The consequence of this is that it will be displayed as such in Dagster's UI, and, if you launch a run to materialize both of them, Dagster won't run `order_count_chart` until `customers` completes.
```

Comment:
Consequence is ambiguous, and I cannot tell if this is a bad thing or good thing. Let's just use a different word.

```suggestion
   The line `deps=get_asset_key_for_model([jaffle_shop_dbt_assets], "customers")` tells Dagster that this asset is downstream of the `customers` dbt model. This dependency will be displayed in Dagster's UI. If you launch a run that materializes both assets, Dagster won't run `order_count_chart` until `customers` completes.
```

Reply:
👍